src/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -305,25 +305,28 @@ public void TestMailDelivery()
             }
         }
 
-        [Fact]
-        public async Task TestMailDeliveryAsync()
+        [Theory]
+        [InlineData("howdydoo")]
+        [InlineData("")]
+        [InlineData(null)]
+        public async Task TestMailDeliveryAsync(string body)
         {
             SmtpServer server = new SmtpServer();
             SmtpClient client = new SmtpClient("localhost", server.EndPoint.Port);
-            MailMessage msg = new MailMessage("foo@example.com", "bar@example.com", "hello", "howdydoo");
+            MailMessage msg = new MailMessage("foo@example.com", "bar@example.com", "hello", body);
             string clientDomain = IPGlobalProperties.GetIPGlobalProperties().HostName.Trim().ToLower();
 
             try
             {
                 Thread t = new Thread(server.Run);
                 t.Start();
-                await client.SendMailAsync(msg);
+                await client.SendMailAsync(msg).TimeoutAfter((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
```

Comment:
Did you confirm that this test will timeout (i.e. fail) without your fix?

Reply:
Yes

Reply:
If you want I'll attach image of failure!Let me know! Thank's for review!